packages/components/src/tooltip/index.tsx

```diff
@@ -8,22 +8,37 @@ import * as Ariakit from '@ariakit/react';
  * WordPress dependencies
  */
 import { useInstanceId } from '@wordpress/compose';
-import { Children } from '@wordpress/element';
+import { Children, cloneElement } from '@wordpress/element';
 import deprecated from '@wordpress/deprecated';
 
 /**
  * Internal dependencies
  */
-import type { TooltipProps } from './types';
+import type { TooltipProps, TooltipInternalContext } from './types';
 import Shortcut from '../shortcut';
 import { positionToPlacement } from '../popover/utils';
+import {
+	contextConnect,
+	useContextSystem,
+	ContextSystemProvider,
+} from '../context';
+import type { WordPressComponentProps } from '../context';
 
 /**
  * Time over anchor to wait before showing tooltip
  */
 export const TOOLTIP_DELAY = 700;
 
-function Tooltip( props: TooltipProps ) {
+const CONTEXT_VALUE = {
+	Tooltip: {
+		isNestedInParentTooltip: true,
```

Comment:
I also think we could use the opportunity to document and explain the rationale behind how we handle nested tooltip behavior in the component README.

Reply:
Added a section in the README [`7fbfcbe` (#57202)](https://github.com/WordPress/gutenberg/pull/57202/commits/7fbfcbef86c32765dbbdfd8bf27ccc60186894b9)